mcs/class/Mono.Data.Tds/Mono.Data.Tds.Protocol/Tds70.cs

```diff
@@ -559,9 +559,8 @@ private void WriteParameterInfo (TdsMetaParameter param)
 			 * If the value is null, not setting the size to 0 will cause varchar
 			 * fields to get inserted as an empty string rather than an null.
 			 */
-			if (colType != TdsColumnType.IntN && colType != TdsColumnType.DateTimeN) {
```

Comment:
Because it did't fix the problem. It only worked with Int and DateTime types. But when Doubles and another types were used the bug was still existing. This check has been appeared after original PR was offered. I think it was incorrect bugfix that fixed only ints and datetimes.
Correct check is written by @egorovmv in his original PR.